test/unit/mongo_logger.test.ts

```diff
@@ -0,0 +1,270 @@
+import { expect } from 'chai';
+
+import { MongoLogger, SeverityLevel } from '../../src/mongo_logger';
+
+describe('class MongoLogger', function () {
+  describe('#constructor', function () {
+    context('when an invalid value is passed', function () {
+      it('sets the log destination to stderr if an invalid value is passed', function () {
+        const loggerOptions = MongoLogger.resolveOptions({ MONGODB_LOG_PATH: 'invalid' }, {});
+        const logger = new MongoLogger(loggerOptions);
+        expect(logger.logDestination).to.equal(process.stderr);
+      });
+    });
+  });
+
+  describe('static #resolveOptions', function () {
+    it('treats severity values as case-insensitive', function () {
+      const loggerOptions = MongoLogger.resolveOptions(
+        {
+          MONGODB_LOG_COMMAND: 'EMERGENCY',
+          MONGODB_LOG_TOPOLOGY: 'critical',
+          MONGODB_LOG_SERVER_SELECTION: 'aLeRt'
+        },
+        {}
+      );
+      expect(loggerOptions).to.have.nested.property(
+        'severitySettings.command',
+        SeverityLevel.EMERGENCY
+      );
+      expect(loggerOptions).to.have.nested.property(
+        'severitySettings.topology',
+        SeverityLevel.CRITICAL
+      );
+      expect(loggerOptions).to.have.nested.property(
+        'severitySettings.serverSelection',
+        SeverityLevel.ALERT
+      );
+    });
+
+    it('treats invalid severity values as off', function () {
+      const loggerOptions = MongoLogger.resolveOptions(
+        {
+          MONGODB_LOG_COMMAND: 'invalid'
+        },
+        {}
+      );
+      expect(loggerOptions).to.have.nested.property('severitySettings.command', SeverityLevel.OFF);
+    });
+
+    it('can set severity levels per component', function () {
+      const loggerOptions = MongoLogger.resolveOptions(
+        {
+          MONGODB_LOG_COMMAND: SeverityLevel.EMERGENCY,
+          MONGODB_LOG_TOPOLOGY: SeverityLevel.CRITICAL,
+          MONGODB_LOG_SERVER_SELECTION: SeverityLevel.ALERT,
+          MONGODB_LOG_CONNECTION: SeverityLevel.DEBUG,
+          MONGODB_LOG_ALL: SeverityLevel.WARNING
+        },
+        {}
+      );
+      expect(loggerOptions).to.have.nested.property(
+        'severitySettings.command',
+        SeverityLevel.EMERGENCY
+      );
+      expect(loggerOptions).to.have.nested.property(
+        'severitySettings.topology',
+        SeverityLevel.CRITICAL
+      );
+      expect(loggerOptions).to.have.nested.property(
+        'severitySettings.serverSelection',
+        SeverityLevel.ALERT
+      );
+      expect(loggerOptions).to.have.nested.property(
+        'severitySettings.connection',
+        SeverityLevel.DEBUG
+      );
+      expect(loggerOptions).to.have.nested.property(
+        'severitySettings.default',
+        SeverityLevel.WARNING
+      );
+    });
+
+    context('when component severities are not set or invalid', function () {
```

Comment:
you are missing a case of what happens when the default severity level (`MONGODB_LOG_ALL`) is also invalid